.github/workflows/ci.yaml

```diff
@@ -209,162 +139,202 @@ jobs:
+    name: Valgrind Test CLN ${{ matrix.name }}
+    runs-on: ubuntu-22.04
     env:
+      COMPAT: 1
+      BITCOIN_VERSION: 0.20.1
+      ELEMENTS_VERSION: 0.18.1.8
+      RUST_PROFILE: release  # Has to match the one in the compile step
+      VALGRIND: 1
+      CFG: gcc-dev1-exp1
       DEVELOPER: 1
-      RUST: 1
-      VALGRIND: 0
-      # Run only the rust tests, others are not impacted.
-      TEST_CMD: "make -j 8 && pytest -vvv tests/test_cln_rs.py"
+      EXPERIMENTAL_FEATURES: 1
+      PYTEST_OPTS: --test-group-random-seed=42
+    needs:
+      - compile
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - NAME: Valgrind (01/10)
+            PYTEST_OPTS: --test-group=1  --test-group-count=10
+          - NAME: Valgrind (02/10)
+            PYTEST_OPTS: --test-group=2  --test-group-count=10
+          - NAME: Valgrind (03/10)
+            PYTEST_OPTS: --test-group=3  --test-group-count=10
+          - NAME: Valgrind (04/10)
+            PYTEST_OPTS: --test-group=4  --test-group-count=10
+          - NAME: Valgrind (05/10)
+            PYTEST_OPTS: --test-group=5  --test-group-count=10
+          - NAME: Valgrind (06/10)
+            PYTEST_OPTS: --test-group=6  --test-group-count=10
+          - NAME: Valgrind (07/10)
+            PYTEST_OPTS: --test-group=7  --test-group-count=10
+          - NAME: Valgrind (08/10)
+            PYTEST_OPTS: --test-group=8  --test-group-count=10
+          - NAME: Valgrind (09/10)
+            PYTEST_OPTS: --test-group=9  --test-group-count=10
+          - NAME: Valgrind (10/10)
+            PYTEST_OPTS: --test-group=10 --test-group-count=10
     steps:
       - name: Checkout
-        uses: actions/checkout@v2.0.0
+        uses: actions/checkout@v3
 
       - name: Set up Python 3.7
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: 3.7
 
       - name: Install dependencies
         run: |
-          bash -x .github/scripts/setup.sh
+          sudo apt-get install -yyq valgrind
+          pip3 install --user pip wheel poetry
+          poetry install
 
-      - name: Build
-        run: |
-          bash -x .github/scripts/build.sh
+      - name: Install bitcoind
+        run: .github/scripts/install-bitcoind.sh
 
-      - name: Upload Unit Test Results
-        if: always()
-        uses: actions/upload-artifact@v2
+      - name: Download build
+        uses: actions/download-artifact@v3
         with:
-          name: Junit Report ${{ github.run_number }}.${{ matrix.cfg }}
-          path: report.*
+          name: cln-gcc-dev1-exp1.tar.bz2
+
+      - name: Unpack build
+        run: tar -xvjf cln-gcc-dev1-exp1.tar.bz2
+
+      - name: Test
+        env:
+          COMPAT: 1
+          SLOW_MACHINE: 1
+          TEST_DEBUG: 1
+        run: |
+
+          sed -i 's/VALGRIND=0/VALGRIND=1/g' config.vars
+          poetry run pytest tests/ -vvv -n 5 ${PYTEST_OPTS}  ${{ matrix.PYTEST_OPTS }}
```

Comment:
We previously used PYTEST_PAR=3 here while valgrind was in use. 5 tests might be too ambitious and responsible for the SIGTERM we're getting on test-group 6.  It looks like these were the active tests when it occurs:
tests/test_connection.py::test_funding_close_upfront 
tests/test_pay.py::test_reject_invalid_payload
tests/test_misc.py::test_address
tests/test_pay.py::test_mpp_presplit
tests/test_closing.py::test_invalid_upfront_shutdown_script

Reply:
You're right, this might be too aggressive and maybe we're running out of memory causing the fault?

Reply:
Reduced to 4, let's see if that works, otherwise I'll reduce it to 3 as well :-)